lab_3/main.py

```diff
@@ -2,8 +2,47 @@
 Lab 3
 Language classification using n-grams
 """
-
+import json
+import math
 from typing import Dict, Tuple
+import re
+
+
+def _split_into_letters(token: str) -> tuple:
+    """
+    Splits a token into letters, framed with '_'
+    :param token: a token
+    :return: a token split into letters
+    """
+    ascii_replacement = {"ä": "ae",
+                         "ö": "oe",
+                         "ü": "ue",
+                         "ß": "ss"}
+    # replace specified non-ascii letters
+    for key, replacement in ascii_replacement.items():
+        token.replace(key, replacement)
+    # split into letters, remove punctuation
+    token = [char for char in token if char.isalpha()]
+    # append framing characters
+    if token:
+        token.insert(0, "_")
+        token.append("_")
+    return tuple(token)
+
+
+def _tokenize(text: str) -> tuple:
```

Comment:
The top-level functions are not usually named like this - the `_` indicates a protected *method*, i.e. it should be a member of a class. 
This does not apply here - use the regular naming.